test/infrastructure/docker/templates/clusterclass-in-memory.yaml

```diff
@@ -108,46 +111,48 @@ spec:
     metadata:
       annotations:
     machineInfrastructure:
-      ref:
-        apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+      templateRef:
+        apiVersion: infrastructure.cluster.x-k8s.io/v1beta2
         kind: DevMachineTemplate
         name: in-memory-control-plane
-    ref:
-      apiVersion: controlplane.cluster.x-k8s.io/v1beta1
+    templateRef:
+      apiVersion: controlplane.cluster.x-k8s.io/v1beta2
       kind: KubeadmControlPlaneTemplate
       name: in-memory-control-plane
     machineHealthCheck:
-      unhealthyConditions:
+      maxUnhealthy: 100%
```

Comment:
```suggestion
```
Should not be necessary. Please also drop the one below

Same in clusterclass-quick-start.yaml